presto-jdbc/src/main/java/com/facebook/presto/jdbc/ConnectionProperties.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.jdbc;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Set;
+
+import static java.lang.String.format;
+
+public class ConnectionProperties
+{
+    public static final ConnectionProperty USER = new User();
+    public static final ConnectionProperty PASSWORD = new Password();
+    public static final ConnectionProperty SSL = new Ssl();
+    public static final ConnectionProperty SSL_TRUST_STORE_PATH = new SslTrustStorePath();
+    public static final ConnectionProperty SSL_TRUST_STORE_PASSWORD = new SslTrustStorePassword();
+
+    public static final Set<ConnectionProperty> allOf = ImmutableSet.of(
+            USER,
+            PASSWORD,
+            SSL,
+            SSL_TRUST_STORE_PATH,
+            SSL_TRUST_STORE_PASSWORD);
+
+    private static final Map<String, ConnectionProperty> keyLookup;
+
+    static {
+        ImmutableMap.Builder<String, ConnectionProperty> builder = ImmutableMap.builder();
+        for (ConnectionProperty property : allOf) {
+            builder.put(property.getKey(), property);
+        }
+        keyLookup = builder.build();
+    }
+
+    private ConnectionProperties()
+    {
+    }
+
+    public static ConnectionProperty forKey(String propertiesKey)
+    {
+        return keyLookup.get(propertiesKey);
+    }
+
+    public static Map<String, String> getDefaults()
+    {
+        ImmutableMap.Builder<String, String> result = ImmutableMap.builder();
+        for (ConnectionProperty property : allOf) {
+            Optional<String> value = property.getDefault();
+            if (value.isPresent()) {
+                result.put(property.getKey(), value.get());
+            }
+        }
+        return result.build();
```

Comment:
After fighting with Java over generics together last night, I have elected to leave this one as-is.